### PR TITLE
Fix broken queries

### PR DESCRIPTION
**Pull request by ammsalme**
1) Fix generating candidate search queries for standard identifiers:

- generate search query for identifiers for all valid subfields of a standard identifier field instead of just two first subfields [MELKEHITYS-1850]
- do not generate search queries with 'undefined' as identifier [MELKEHITYS-1852]
- do not generate duplicate search queries (even if same identifier string is found multiple times in record) [MELKEHITYS-1854]
- generate search query only for valid identifier strings for other fields (non 020/022) [MELKEHITYS-1848]

2) Add debugData to use '**:data**' namespace for very verbose debug output [MELKEHITYS-1843]

### CODE REVIEW

File: src/candidate-search/query-list/bib.js [hunk 1]
```diff
@@ -83,21 +83,47 @@ export function bibTitle(record) {
...
   const fields = record.get(/^(?<def>020|022|024)$/u);
-  const identifiers = fields.map(toIdentifiers);
-  const pairs = identifiers.reduce(toPairs, []);
+  const identifiers = [].concat(...fields.map(toIdentifiers));
+  const uniqueIdentifiers = [...new Set(identifiers)];
 
-  return pairs.map(([a, b]) => b ? `dc.identifier=${a} or dc.identifier=${b}` : `dc.identifier=${a}`);
+  debug(`Identifiers (${identifiers.length}): ${JSON.stringify(identifiers)}`);
+  debug(`Standard identifier fields: ${JSON.stringify(fields)}`);
+  debug(`Unique identifiers (${uniqueIdentifiers.length}): ${JSON.stringify(uniqueIdentifiers)}`);
```
Comment (natlibfi-jonollil) [inline src/candidate-search/query-list/bib.js:98]: Debugs after length check and debug note when no identifiers are found?

File: src/candidate-search/query-list/bib.js [hunk 2]
```diff
@@ -83,21 +83,47 @@ export function bibTitle(record) {
...
+      if (array.length === 0) {
+        return [];
+      }
+      return [array.slice(0, 2)].concat(toPairs(array.slice(2), 2));
+    }
+  }
 
   function toIdentifiers({tag, subfields}) {
     const issnIsbnReqExp = (/^[A-Za-z0-9-]+$/u);
+    const otherIdReqExp = (/^[A-Za-z0-9-:]+$/u);
```
Comment (natlibfi-jonollil) [inline src/candidate-search/query-list/bib.js:126]: if used reqexp is identical to issnIsbnReqExp rename to more common style name and use it to all?

File: test-fixtures/candidate-search/query-list/bib/standardIdentifiers/06/metadata.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {
-  "description": "DISABLED: Should not generate 'undefined' standard identifier query (ISSN with colon)",
-  "enabled": false,
+  "description": "Should not generate 'undefined' standard identifier query (ISSN with colon)",
+  "enabled": true,
```
Comment (natlibfi-jonollil) [inline test-fixtures/candidate-search/query-list/bib/standardIdentifiers/06/metadata.json:3]: No need to add this if there is enabled = true default value in fixugen callback

**Review by natlibfi-jonollil — CHANGES_REQUESTED** [review]

File: src/candidate-search/query-list/bib.js (hunk 2 above)
Comment (ammsalme) [reply src/candidate-search/query-list/bib.js:126]: ReqExp is **not** identical, other IDs can include colon **:** which is not valid for ISBNs and ISSNs.

File: src/candidate-search/query-list/bib.js (hunk 1 above)
Comment (ammsalme) [reply src/candidate-search/query-list/bib.js]: Added debug note for case of no identifiers, moved verbose debug data to ':data' namespace.

File: test-fixtures/candidate-search/query-list/bib/standardIdentifiers/06/metadata.json (hunk 3 above)
Comment (ammsalme) [reply test-fixtures/candidate-search/query-list/bib/standardIdentifiers/06/metadata.json]: Removed.

**Review by natlibfi-jonollil — APPROVED** [review]

**Pull request merged**